avm/utilities/pipelines/staticValidation/psrule/ps-rule.yaml

```diff
@@ -76,5 +76,4 @@ rule:
   includeLocal: false
   exclude:
     # Ignore the following rules for all resources
-    - Azure.KeyVault.PurgeProtect
```

Comment:
We cannot remove this as we need purge protection to be disabled on keyvault in order to be able to continuously deploy and delete the resource but also be able to run idempotency test on the same resource

Reply:
okay understood. But we are saying that the module defaults should be WAF aligned, so how do we achieve that?

Our tests shouldnt block us from achieving this.

Reply:
I see the angle and I can assure this has been discussed many times in the past as well.
I'm open to further discuss that, so far we did not find a valid alternative to mitigate this issue.

Purge protection for our CI cannot be enforced until we are able to automatically recover soft-deleted resources. That will not only solve this blocker, but also simplify a lot our CI removal process, currently taking into consideration all soft deleted resources and making sure they are purged for the above reason.

I'd be inclined to have purge protection as an exception for the moment, until we agree on the alternative, since this may take some time to be implemented and we shouldn't block keyvault e2e validation meanwhile. This may also be the case for other resources as well.
@AlexanderSehr @ChrisSidebotham what's your take on that?

Reply:
I agree with the point your making @eriqua, until we can recover vaults this will cause a headache. Maybe something for us to align with PG on. for example Terraform handles this by recovering the purged vault on "Creation of the Resource" albeit this causes some confusion but would be nice to be handled at the layer above

Reply:
okay so if i added the exclusion back for the key vault are we good to progress with this PR?

Reply:
For this comment yes, I'll leave the resolution of the other one to @rahalan

Reply:
I see no reason why not @eriqua ?

Reply:
I swear I've seen a soft-deleted resource recovery even for at least one ARM resource (kvlt) - can't remember which one though. @jtracey93 if you know somebody who knows somebody that can make this happen for kvlt, that would be great - yet I assume the PGs knows about this request and just did not implement it (yet).

In any case, the only way to get around this for now would be to autogenerate the name in the test run. Unless you run the test file deployment twice, this would work even with purge protection (like we did for all the encryption key vaults). However, we're already collecting key vaults like candy. @eriqua would you know if we can keep the implement in the test files as is but add a supression in the test files e.g. above the name parameter? This would enable us to remove the rule from the central file. But I'm not sure if PSRule supports this.

Reply:
Let's discuss alternatives to purge enablement offline and I'd agree to merge the remaining changes meanwhile.